physics/sfcsub.F

```diff
@@ -2019,12 +2019,14 @@ subroutine sfccycle(lugb,len,lsoil,sig1t,deltsfc                    &
 !    *,' tsffcs=',tsffcs(iprnt),' slianl=',slianl(iprnt)
 
       do i=1,len
+       if (nint(slmskl(i)) /= 1) then
         if (sicanl(i) >= min_ice(i)) then
```

Comment:
I think that with a fractional grid it is better not to use slmsk.

Reply:
@DomHeinzeller Dom, slmskl is computed in gcycle.F90, and this computation looks questionable to me. But I assume that people who wrote the code tested it with frac_grid=.true. Therefore, I approved this PR.

Reply:
@smoorthi-emc wrote the code and I am highly confident that he tested the logic with fractional grid turned on. Moorthi, can you confirm that the above code change in `sfcsub.F` works with fractional grid as intended? Thanks!